src/DataStructures/Python/DataVector.cpp

```diff
@@ -40,6 +61,48 @@ void bind_datavector() {
       .def(bp::self += bp::other<double>{})
       // Need to do math explicitly converting to DataVector because we don't
       // want to represent all the possible expression template types
+      .def("abs", +[](const DataVector& t) { return DataVector{abs(t)}; })
+      .def("acos", +[](const DataVector& t) { return DataVector{acos(t)}; })
+      .def("acosh", +[](const DataVector& t) { return DataVector{acosh(t)}; })
+      .def("asin", +[](const DataVector& t) { return DataVector{asin(t)}; })
+      .def("asinh", +[](const DataVector& t) { return DataVector{asinh(t)}; })
+      .def("atan", +[](const DataVector& t) { return DataVector{atan(t)}; })
+      .def("atan2",
+           +[](const DataVector& y, const DataVector& x) {
+             return DataVector{atan2(y, x)};
+           })
+      .def("atanh", +[](const DataVector& t) { return DataVector{atanh(t)}; })
+      .def("cbrt", +[](const DataVector& t) { return DataVector{cbrt(t)}; })
+      .def("cos", +[](const DataVector& t) { return DataVector{cos(t)}; })
+      .def("cosh", +[](const DataVector& t) { return DataVector{cosh(t)}; })
+      .def("erf", +[](const DataVector& t) { return DataVector{erf(t)}; })
+      .def("erfc", +[](const DataVector& t) { return DataVector{erfc(t)}; })
+      .def("exp", +[](const DataVector& t) { return DataVector{exp(t)}; })
+      .def("exp2", +[](const DataVector& t) { return DataVector{exp2(t)}; })
+      .def("exp10", +[](const DataVector& t) { return DataVector{exp10(t)}; })
+      .def("fabs", +[](const DataVector& t) { return DataVector{fabs(t)}; })
+      .def("hypot",
+           +[](const DataVector& x, const DataVector& y) {
+             return DataVector{hypot(x, y)};
+           })
+      .def("invcbrt",
+           +[](const DataVector& t) { return DataVector{invcbrt(t)}; })
+      .def("invsqrt",
+           +[](const DataVector& t) { return DataVector{invsqrt(t)}; })
+      .def("log", +[](const DataVector& t) { return DataVector{log(t)}; })
+      .def("log2", +[](const DataVector& t) { return DataVector{log2(t)}; })
+      .def("log10", +[](const DataVector& t) { return DataVector{log10(t)}; })
+      .def("max", +[](const DataVector& t) { return double{max(t)}; })
+      .def("min", +[](const DataVector& t) { return double{min(t)}; })
+      .def("pow", +[](const DataVector& base,
```

Comment:
This makes `dv.pow(3)` work, but you should add a `__pow__` function so that `dv**3` does the same.

Reply:
Is this true for all the functions? We should probably have tests that check `dv.blah()` and `blah(dv)` works.

Reply:
No, a lot of functions such as `min(dv)` and `max(dv)` work already because `__iter__` is implemented. We only need `__pow__` to support the `**` operator. There's also a lot more special methods we could add (here's a list: https://rszalski.github.io/magicmethods/), but it's probably easier and sufficient to rely on numpy.